parser/src/parser/grammar.lalrpop

```diff
@@ -87,8 +87,13 @@ TransitionConstraint: TransitionConstraint = {
 // ================================================================================================
 
 Expr: Expr = {
-    <lexpr: Expr> "+" <rexpr: NextExpr> => Expr::Add(Box::new(lexpr), Box::new(rexpr)),
-    <lexpr: Expr> "-" <rexpr: NextExpr> => Expr::Subtract(Box::new(lexpr), Box::new(rexpr)),
+    <lexpr: Expr> "+" <rexpr: ExpExpr> => Expr::Add(Box::new(lexpr), Box::new(rexpr)),
+    <lexpr: Expr> "-" <rexpr: ExpExpr> => Expr::Subtract(Box::new(lexpr), Box::new(rexpr)),
+    ExpExpr
+}
+
+ExpExpr: Expr = {
+    <lexpr: NextExpr> "^" <rexpr: NextExpr> => Expr::Exp(Box::new(lexpr), Box::new(rexpr)),
```

Comment:
I don't think we want to allow a `NextExpr` on the right hand side here. Even when we support variables in the future, we don't want to allow trace columns here. I think this is actually the best place to restrict the rhs to a constant - then you don't have to check it later, since it will be handled during parsing.

Reply:
Makes sense. Thanks.